src/digger/digger/digger_node.py

```diff
@@ -182,23 +173,20 @@ def lift_set_power_callback(self, request, response):
     def zero_lift_callback(self, request, response):
         """This service request zeros the lift system."""
         self.zero_lift()
+        while not self.top_limit_pressed:
+            pass
         response.success = 0  # indicates success
         return response
 
-    # Define timer callback methods here
-    def timer_callback(self):
-        """Publishes whether or not the current goal position has been reached."""
-        # This service call will return a future object, that will eventually contain the position in degrees
-        future = self.cli_motor_get.call_async(MotorCommandGet.Request(type="position", can_id=self.DIGGER_LIFT_MOTOR))
-        future.add_done_callback(self.done_callback)
-
-    def done_callback(self, future):
-        self.current_position_degrees = future.result().data
-        goal_reached_msg = Bool(
-            data=abs(self.current_goal_position + self.lift_encoder_offset - self.current_position_degrees)
-            <= self.goal_threshold
-        )
-        self.publisher_goal_reached.publish(goal_reached_msg)
+    def lower_lift_callback(self, request, response):
+        """This service request reverse-zeros the lift system, putting it at the lowest point"""
+        self.lower_lift()
+        while not self.bottom_limit_pressed:
+            pass
```

Comment:
This is burning cpu cycles, create an event that will trigger instead of spinning a while loop constantly

Reply:
Can you explain more about how you might go about this? Would take more research (Me, Anthony, and Alex dont know)

Reply:
Use a python event and trigger it. So you create a global event in the node then wait for it here and trigger it when the limit switch is pressed.
there is something similar here, but better examples can be found online
https://github.com/GOFIRST-Robotics/Lunabotics/blob/main/src/gstreamer/gstreamer/client_gstreamer.py#L82